packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2587,7 +2587,8 @@ export class KeyringController extends BaseController<
     const accounts = await this.#getAccountsFromKeyrings();
 
     switch (type) {
-      case KeyringTypes.simple: {
+      case KeyringTypes.simple:
+      case KeyringTypes.hd: {
```

Comment:
Just for the record:
- Given that we only check for duplicates when creating new keyrings, this shouldn't impact the "manual" HD keyring account creation.
- For multi-SRP, this will prevent duplicated addresses when importing a new SRP.